router.go

```diff
@@ -36,19 +37,27 @@ type serviceProcedure struct {
 	procedure string
 }
 
+type serviceProcedureEncoding struct {
+	service   string
+	procedure string
+	encoding  transport.Encoding
+}
+
 // MapRouter is a Router that maintains a map of the registered
 // procedures.
 type MapRouter struct {
-	defaultService string
-	entries        map[serviceProcedure]transport.Procedure
+	defaultService            string
+	serviceProcedures         map[serviceProcedure]transport.Procedure
+	serviceProcedureEncodings map[serviceProcedureEncoding]transport.Procedure
```

Comment:
This seems a bit convoluted. Perhaps it will be easier to do,

    map[serviceProcedure]encodingProcedure

Where

    type encodingProcedure interface {
        GetProcedure(transport.Encoding) transport.Procedure
    }

With two implementations,

    type wildcardEncodingProcedure struct{ transport.Procedure }
    type perEncodingProcedure map[transport.Encoding]transport.Procedure

This also makes supportedEncodings unnecessary.

You can add other helper functions like `ListProcedures() []transport.Procedure` (for `Procedures()`) as needed.